cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.DnsNameGenerator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Creates HTTP client and interacts with Kafka Agent's REST endpoint
+ */
+class KafkaAgentClient {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaAgentClient.class.getName());
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final String BROKER_STATE_REST_PATH = "/v1/broker-state/";
+    private static final int BROKER_STATE_HTTPS_PORT = 8443;
+
+    private String namespace;
+    private Reconciliation reconciliation;
+    private String cluster;
+    private Secret clusterCaCertSecret;
+    private Secret coKeySecret;
+    private HttpClient httpClient;
+
+    KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
+        this.reconciliation = reconciliation;
+        this.cluster = cluster;
+        this.namespace = namespace;
+        this.clusterCaCertSecret = clusterCaCertSecret;
+        this.coKeySecret = coKeySecret;
+        this.httpClient = createHttpClient();
+    }
+
+    //visible for testing
+    KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace) {
```

Comment:
We usually do something like this when you need to expose some method for testing:

```suggestion
    /* test */ KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace) {
```